### PR TITLE
Testing with RSpec

### PR DESCRIPTION
**Pull request by Stricks1**
On this branch, we added the RSpec file to test our methods for enumerable.

### CODE REVIEW

File: spec/enumerables_spec.rb [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+    it "multiply elements to each other: test parameter" do
+      expect((5..10).my_inject(1, :*)).to eql((5..10).inject(1, :*))
+    end
+
+    it "multiply elements to each other: test block" do
+      expect((5..10).my_inject(1) { |product, n| product * n }).to eql((5..10).inject(1) { |product, n| product * n })
+    end
+
+    it "find the longuer words" do
```
Comment (shshamim63) [inline spec/enumerables_spec.rb]: Do you mean longer?

File: spec/enumerables_spec.rb [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    end
+
+    it "To check the return if a parameter object is given" do
+      expect([1, 2].my_all?(1)).to eql([1, 2].all?(1))
+    end
+
+    it "To check the return if a regular expression is given" do
+      expect(words.my_all?(/d/)).to eql(words.all?(/d/))
+    end
+  end
```
Comment (shshamim63) [inline spec/enumerables_spec.rb]: Add the following test case
```ruby
context 'when no block or argument is given' do
  it 'returns true when none of the collection members are false or nil' do
    expect(ary.my_all?).to eql(ary.all?)
  end
end
```

File: spec/enumerables_spec.rb [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    end
+
+    it "To check the return if a parameter object is given" do
+      expect([1, 2].my_any?(1)).to eql([1, 2].any?(1))
+    end
+
+    it "To check the return if a regular expression is given" do
+      expect(words.my_any?(/d/)).to eql(words.any?(/d/))
+    end
+  end
```
Comment (shshamim63) [inline spec/enumerables_spec.rb]: Add the following test case
```ruby
context 'when no block or argument is given' do
  it ' returns true if at least one of the collection is not false or nil' do
    expect(ary.my_any?).to eql(ary.any?)
  end
end
```

File: spec/enumerables_spec.rb [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    end
+
+    it "To check the return if a parameter object is given" do
+      expect([1, 2].my_none?(1)).to eql([1, 2].none?(1))
+    end
+
+    it "To check the return if a regular expression is given" do
+      expect(words.my_none?(/d/)).to eql(words.none?(/d/))
+    end
+  end
```
Comment (shshamim63) [inline spec/enumerables_spec.rb]: Add the following test case
```ruby
context 'when no block or argument is given' do
  it ' returns true only if none of the collection members is true' do
    false_array = [false, nil, false]
    expect(false_array.my_none?).to eql(false_array.none?)
  end
end
```

**Review by shshamim63 — APPROVED** [review]
## Status :100: Approved :+1: 
___
Hi @Stricks1 and @rindrajosia :wave:
Well done, you have compared your methods with the actual one. But there are some minor issues left. Check my comments below to resolve them.
___
### :spades: Mandatory Changes :spades: 
___
1. I can not see any info related to the test in your readme. So please some info related to testing.
2. You forgot to add the following corner cases for my_all?, my_any?, my_none? methods. Please check the comments attached to the blocks.
3. There is a slight typo mistake.
___
#### As the issues are easy to resolve I am marking it as approved in advance. Please Resolve the issues and then you can merge it.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain
